src/main/java/com/commercetools/paymenttoorderprocessor/jobs/actions/OrderCreator.java

```diff
@@ -66,19 +75,18 @@ public void write(List<? extends CartAndMessage> items) {
 
     private void sendRequestToCreateOrder(CartAndMessage cartAndMessage) {
         final Cart cart = cartAndMessage.getCart();
-        final String body = SphereJsonUtils.toJsonString(cart.getId());
 
-        final String bodyEncrypt = encrypt(body);
-        if (bodyEncrypt == null) {
+        final String encryptedCartId = encrypt(cart.getId());
+        if (encryptedCartId == null) {
             timeStampManager.processingMessageFailed();
             return;
         }
-        final List<NameValuePair> headerList = new ArrayList<NameValuePair>();
-        headerList.add(NameValuePair.of(HttpHeaders.CONTENT_TYPE, "text/plain"));
-        headerList.add(NameValuePair.of("Content-Length", String.valueOf(bodyEncrypt.length())));
-        final HttpHeaders httpHeader = HttpHeaders.of(headerList);
-        final HttpRequestBody httpBody = StringHttpRequestBody.of(bodyEncrypt);
-        final HttpRequest httpRequest = HttpRequest.of(HttpMethod.POST, urlstring, httpHeader, httpBody);
+
+        QueryStringEncoder queryStringEncoder = new QueryStringEncoder(urlstring);
```

Comment:
This is not completely correct:
here `QueryStringEncoder` is used as a simple URL builder, which handles query arguments building (incapsulating URL encoding step, of course). If i use base64 it will look like this:

  ```
  QueryStringEncoder queryStringEncoder = new QueryStringEncoder(urlstring);
  queryStringEncoder.addParam("encryptedCartId", new String(Base64.getUrlEncoder().encode(encryptedCartId.getBytes())));
  ```